src/scripts/main.js

```diff
@@ -1,3 +1,247 @@
+    const cellA = rowA.cells[index].innerText;
+    const cellB = rowB.cells[index].innerText;
+
+    return cellA.localeCompare(cellB, 'en', { numeric: true });
+  });
+
+  if (JSON.stringify(rows.map(row => row.innerText))
+    === JSON.stringify(currentRows.map(row => row.innerText))) {
+    rows.sort((rowA, rowB) => {
+      const cellA = rowA.cells[index].innerText;
+      const cellB = rowB.cells[index].innerText;
+
+      return cellB.localeCompare(cellA, 'en', { numeric: true });
+    });
+  }
+  return rows;
+}
+
+tbody.addEventListener('click', (event) => {
+  if (!event.target.parentElement.tagName === 'TR') {
+    return;
+  }
+
+  const activeClasses = tbody.querySelectorAll('.active');
+  for (const activeClass of activeClasses) {
+    activeClass.classList.remove('active');
+  }
+
+  event.target.parentElement.className = 'active';
+})
+
+tbody.addEventListener('dblclick', (event) => {
+  if (!event.target.tagName === 'TD') {
+    return;
+  }
+
+  const defaultValue = event.target.textContent.trim();
+
+  const inputElement = document.createElement('input');
+  inputElement.type = 'text';
+  inputElement.classList.add('cell-input');
+  inputElement.style.width = defaultValue.length * 10 + 'px';
+  inputElement.value = defaultValue;
+
+  event.target.textContent = '';
+  event.target.append(inputElement);
+  inputElement.focus();
+
+  inputElement.addEventListener('blur', () => {
+    if (inputElement.value === '') {
+      inputElement.parentElement.textContent = defaultValue;
+    } else {
+      inputElement.parentElement.textContent = inputElement.value;
+    }
+  })
+
+  inputElement.addEventListener('keypress', (event) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      if (inputElement.value === '') {
+        inputElement.parentElement.textContent = defaultValue;
+      } else {
+        inputElement.parentElement.textContent = inputElement.value;
+      }
+    }
+  })
+})
+
+function employeeForm() {
+  const form = document.createElement('form');
+  form.classList.add('new-employee-form');
+
+  const nameInput = createInput('name');
+  createLabel('Name:', nameInput, form);
+
+  const positionInput = createInput('position');
+  createLabel('Position:', positionInput, form);
+
+  const officeSelect = createSelect('office', officeOptions);
+  createLabel('Office:', officeSelect, form);
+
+  const ageInput = createInput('age', 'number');
+  createLabel('Age:', ageInput, form);
+
+  const salaryInput = createInput('salary', 'number');
+  createLabel('Salary:', salaryInput, form);
+
+  const saveButton = document.createElement('button');
+  saveButton.type = 'submit';
+  saveButton.textContent = 'Save to table';
+
+  form.addEventListener('submit', (event) => {
+    event.preventDefault();
+
+    const newEmployeeData = [...form.elements];
+    const newRow = document.createElement('tr');
+    let index = 0;
+    const newRowData = {};
+
+    for (const element of newEmployeeData) {
```

Comment:
Replace for of loop with forEach, e.g. newEmployeeData.forEach(element => { ... })